ssh_automated_tests/core/shell.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from time import sleep
+
+from ssh_automated_tests.core.connection import Connection
+
+
+class Shell(ABC):
+    @abstractmethod
+    def execute(self, command: str):
+        pass
+
+    @abstractmethod
+    def execute_with_output(self, command: str):
+        pass
+
+    @abstractmethod
+    def close(self):
+        pass
+
+
+class InteractiveShell(Shell):
```

Comment:
There are several methods which aren't declared in the interface `Shell`. We have to avoid that. Please redesign the interface or the class.